app/jobs/bulk_invoice_job.rb

```diff
@@ -0,0 +1,41 @@
+class BulkInvoiceJob
+  include WickedPdf::PdfHelper
+
+  def initialize(order_ids, directory, filename)
+    @order_ids = order_ids
+    @directory = directory
+    @filename = filename
+  end
+
+  def perform
+    pdf = CombinePDF.new
+    orders = Spree::Order.where(id: @order_ids)
+
+    orders.each do |order|
+      invoice = renderer.render_to_string pdf: "invoice-#{order.number}.pdf",
+                                          template: invoice_template,
+                                          formats: [:html], encoding: "UTF-8",
+                                          locals: { :@order => order }
+
+      pdf << CombinePDF.parse(invoice)
+    end
+
+    pdf.save "#{file_directory}/#{@filename}.pdf"
```

Comment:
I think you need a invoice service with:
- all logic related to filenames, directories and filepaths used in both bulk_invoice_job and in invoices_controller
- invoice_template used in both bulk_invoice_job and orders_controller_decorator

Reply:
Is the idea of services that they have one callable method? I think I'd need several callable methods to allow both `BulkInvoiceJob` and `InvoicesController` to access what they need. Is that ok?

Reply:
It's a very good question!

In general I'd say several methods is fine as long as you keep the service with a single responsibility, because these methods will be very simple I'd put them in an invoices_services.

I think making it mandatory for every service to have a "def call" and only one callable method is too strict and doesnt help, in this case creating 2 services is too much in my opinion.

what do you think @sauloperez? :-)

Reply:
Ok, I've added a controversial service. :grimacing:

Reply:
it's so much better!!!